app/utils/webhooks.py

```diff
@@ -75,13 +75,35 @@ async def _get_sticker_embed(sticker: discord.StickerItem) -> discord.Embed:
     )
 
 
-def _format_subtext(executor: discord.Member | None, msg_data: MessageData) -> str:
+def dynamic_timestamp(dt: dt.datetime, fmt: str | None = None) -> str:
+    fmt = f":{fmt}" if fmt is not None else ""
+    return f"<t:{int(dt.timestamp())}{fmt}>"
+
+
+def _format_subtext(
+    message: discord.Message,
+    executor: discord.Member | None,
+    msg_data: MessageData,
+    *,
+    include_timestamp: bool = True,
+) -> str:
     lines: list[str] = []
     if reactions := msg_data.reactions.items():
         lines.append("   ".join(f"{emoji} x{count}" for emoji, count in reactions))
+    if message.created_at > dt.datetime.now(tz=dt.UTC) - dt.timedelta(hours=12):
+        include_timestamp = False
+    if include_timestamp:
+        line = dynamic_timestamp(message.created_at)
+        if message.edited_at is not None:
+            line += f" (edited at {dynamic_timestamp(message.edited_at, 't')})"
+        lines.append(line)
```

Comment:
Let's make `created_at` and `edited_at` fields of `MessageData` so that we don't have to pass the `discord.Message` separately.